core/src/core/transaction.rs

```diff
@@ -431,14 +431,16 @@ impl Transaction {
 	/// Validates all relevant parts of a fully built transaction. Checks the
 	/// excess value against the signature as well as range proofs for each
 	/// output.
-	pub fn validate(&self) -> Result<(), Error> {
+	pub fn validate(&self, as_block: bool) -> Result<(), Error> {
```

Comment:
This bool flag feels awkward but I think you're tracking this as subsequent cleanup/refactoring in #1333?

Reply:
Correct, at least that was the hope. Yes, slightly awkward but at the moment removes a bunch of duplication.